### ISSUE TITLE
opening_brace is warning about braces within regex literals

### ISSUE THREAD

**Issue by grantneufeld** [realm/SwiftLint#5452]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When a regex literal contains braces (e.g., `/.{3}/`, the `opening_brace` rule is reporting warnings. But, in regex spaces impact the behaviour, so should not follow the same rules as Swift.

Proposed fix: have `opening_brace` ignore regex literals.

### Environment

* SwiftLint: 0.53.0
* Homebrew (running as Run Script phase in Build Phases)
* Configuration file leaves defaults of `opening_brace` unchanged.
* Xcode: 15.2

### Sample

The following code:

```swift
let example = "An example string"
let matches = example.wholeMatch(of: /([A-Z][a-z]+ ){2}[a-z]+/)
```

produces this SwiftLint warning:

“Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration (opening_brace)”

Because the regex has the `{2}` part (which should not have the spaces SwiftLint is expecting).

**Comment by SimplyDanny** [comment]
This has already been fixed by #5164.

**Comment by grantneufeld** [comment]
Thanks!

Will there be a new version released soon that incorporates that fix?

I also note that Homebrew is still at version 0.53.0, but it seems 0.54.0 was released here a few months ago.

**Comment by SimplyDanny** [comment]
> Will there be a new version released soon that incorporates that fix?

There's none scheduled. Quite a few things have happened since 0.54.0, however. So there could indeed be a new one soon. Kindly pinging @jpsim. 😉 

> I also note that Homebrew is still at version 0.53.0, but it seems 0.54.0 was released here a few months ago.

A new Homebrew release needs to wait for version 5.10 of the Swift compiler. Currently, SwiftLint [doesn't build](https://github.com/Homebrew/homebrew-core/pull/153931) in a Homebrew context due to sandboxing issues.
